static/js/components/SiteContent.tsx

```diff
@@ -0,0 +1,140 @@
+import React from "react"
+import { Modal, ModalBody, ModalHeader } from "reactstrap"
+import { useMutation, useRequest } from "redux-query-react"
+import { useSelector } from "react-redux"
+import { FormikHelpers } from "formik"
+
+import {
+  createWebsiteContentMutation,
+  editWebsiteContentMutation,
+  EditWebsiteContentPayload,
+  NewWebsiteContentPayload,
+  websiteContentDetailRequest
+} from "../query-configs/websites"
+import { getWebsiteContentDetailCursor } from "../selectors/websites"
+import SiteContentForm, {
+  splitFieldsIntoColumns
+} from "./forms/SiteContentForm"
+import { contentFormValuesToPayload } from "../lib/site_content"
+import { getResponseBodyError, isErrorResponse } from "../lib/util"
+
+import { EditableConfigItem, Website } from "../types/websites"
+import { ContentFormType } from "../types/forms"
+
+type SiteFormValues = {
+  [key: string]: string
+}
+
+interface Props {
+  uuid: string | null
+  visibility: boolean
+  toggleVisibility: () => void
+  site: Website
+  configItem: EditableConfigItem
+  formType: ContentFormType
+  contentType: string
+  repeatListingRequest: () => void
+}
+
+export default function SiteContent(props: Props): JSX.Element | null {
+  const {
+    visibility,
+    configItem,
+    uuid,
+    toggleVisibility,
+    site,
+    formType,
+    contentType,
+    repeatListingRequest
+  } = props
+
+  const [
+    { isPending: addIsPending },
+    addWebsiteContent
+  ] = useMutation((payload: NewWebsiteContentPayload) =>
+    createWebsiteContentMutation(site.name, payload)
+  )
+  const [{ isPending: editIsPending }, editWebsiteContent] = useMutation(
+    (payload: EditWebsiteContentPayload | FormData) =>
+      // @ts-ignore: uuid and configItem should never be null if editing
```

Comment:
can we get around the null error with the bang operator (`!`) instead of `@ts-ignore`?